cql3/statements/select_statement.cc

```diff
@@ -551,7 +551,7 @@ indexed_table_select_statement::do_execute_base_query(
             concurrency *= 2;
             return proxy.query(_schema, command, std::move(prange), options.get_consistency(), {timeout, state.get_permit(), state.get_client_state(), state.get_trace_state()})
             .then([&ranges_to_vnodes, &merger] (service::storage_proxy::coordinator_query_result qr) {
-                bool is_short_read = qr.query_result->is_short_read();
+                auto is_short_read = qr.query_result->is_short_read();
```

Comment:
Why not add a regular, implicit conversion from bool_class to bool?
We of course won't have the opposite direction (the conversion of bool to bool_class is explicit, that's the whole point of bool_class), but why not allow implicit conversion from bool_class to bool?

Reply:
I don't know. Maybe it's worthwhile. I think implicit conversions are dangerous and it's better not to allow them. Maybe an implicit conversion to bool will also convert to other integrals.

Reply:
Well, somehow (?) we did have an implicit conversion to bool in gcc, and nobody complained, and your patch doesn't get rid of it. But never mind, your patch is fine too. We can rethink this later. I'll resolve this conversation.